search/tasks_test.py

```diff
@@ -40,3 +56,180 @@ def test_increment_document_integer_field_task(mocked_api):
     increment_document_integer_field(*indexing_api_args)
     assert mocked_api.increment_document_integer_field.call_count == 1
     assert mocked_api.increment_document_integer_field.call_args[0] == indexing_api_args
+
+
+def test_wrap_retry_exception():
+    """wrap_retry_exception should raise RetryException when other exceptions are raised"""
+    with wrap_retry_exception(KeyError):
+        # Should not raise an exception
+        pass
+
+
+@pytest.mark.parametrize("matching", [True, False])
+def test_wrap_retry_exception_matching(matching):
+    """A matching exception should raise a RetryException"""
+
+    class SubError(KeyError):
+        """Use a subclass to assert isinstance use"""
+
+    def raise_thing():
+        """raise the exception"""
+        if matching:
+            raise SubError()
+        else:
+            raise TabError()
+
+    matching_exception = RetryException if matching else TabError
+    with pytest.raises(matching_exception):
+        with wrap_retry_exception(KeyError):
+            raise_thing()
+
+
+def test_index_post(mocker):
+    """index_post should call the api function of the same name"""
+    index_post_mock = mocker.patch('search.indexing_api.index_post')
+    wrap_mock = mocker.patch('search.tasks.wrap_retry_exception')
+    api_username = 'username'
+    post_id = 'post_id'
+    index_post.delay(api_username, post_id)
+
+    index_post_mock.assert_called_once_with(api_username, post_id)
+    wrap_mock.assert_called_once_with(PrawcoreException, PRAWException)
+
+
+def test_index_channel(mocker):
+    """index_channel should all posts of a channel"""
+    user = UserFactory.create()
+    index_post_mock = mocker.patch('search.tasks.index_post', autospec=True)
+    api_mock = mocker.patch('channels.api.Api', autospec=True)
+
+    expected_exception = ZeroDivisionError
```

Comment:
Can you explain the need for this?

Reply:
The Celery tasks are chained together, and to ensure that the parent task waits for all the other child tasks you can `raise self.replace(new_task_here)`. That will tell Celery to stop running code from the parent task and start running code from the child task(s), and it will also keep the same task id, which is important for other code waiting on it.

The Celery docs recommend `raise self.replace(...)` over just `self.replace(...)` because it tells the reader that the code ends there: http://docs.celeryproject.org/en/master/reference/celery.app.task.html#celery.app.task.Task.replace

Should I add a comment explaining that? I can add one at least for `ZeroDivisionError` to say that the exception doesn't matter.